### ISSUE TITLE
Tidy up the `docs` folder

### ISSUE THREAD

**Issue by antgamdia** [kubeapps/kubeapps#4123]
### Description:

As Kubeaps is rapidly evolving, some documentation is becoming outdated. Besides, the current  `docs` folder structure is not the best.

We should come up with a way to quickly be able to look up what we want to; for instance, using more "how-to" docs or following a different folder structure.

This is also related to the #3953 .

**Comment by antgamdia** [comment]
For the record, here is the list of current doc files we have:

```
./chart/kubeapps/README.md

./docs/user/dashboard.md
./docs/user/using-an-OIDC-provider.md
./docs/user/offline-installation.md
./docs/user/access-control.md
./docs/user/getting-started.md
./docs/user/operators.md
./docs/user/deploying-to-multiple-clusters.md
./docs/user/syncing-apprepository-webhook.md
./docs/user/using-an-OIDC-provider-with-pinniped.md
./docs/user/private-app-repository.md
./docs/user/OIDC/OAuth2OIDC-oauth2-proxy.md
./docs/user/OIDC/OAuth2OIDC-VMware-cloud-services.md
./docs/user/OIDC/OAuth2OIDC-dex.md
./docs/user/OIDC/OAuth2OIDC-azure-active-directory.md
./docs/user/OIDC/OAuth2OIDC-debugging.md
./docs/user/OIDC/OAuth2OIDC-keycloak.md
./docs/user/OIDC/OAuth2OIDC-google-openid-connect.md

./docs/step-by-step/kubeapps-on-tkg/conclusion.md
./docs/step-by-step/kubeapps-on-tkg/introduction.md
./docs/step-by-step/kubeapps-on-tkg/step-4.md
./docs/step-by-step/kubeapps-on-tkg/README.md
./docs/step-by-step/kubeapps-on-tkg/step-2.md
./docs/step-by-step/kubeapps-on-tkg/step-3.md
./docs/step-by-step/kubeapps-on-tkg/step-1.md

./docs/architecture/overview.md
./docs/architecture/design-proposals/clarity-ui.md
./docs/architecture/design-proposals/authentication-and-authorization.md
./docs/architecture/design-proposals/multi-cluster-support.md
./docs/architecture/design-proposals/OCI-registries.md
./docs/architecture/design-proposals/operators-support-poc.md
./docs/architecture/design-proposals/helm3.md
./docs/architecture/design-proposals/README.md
./docs/architecture/design-proposals/dashboard/deployment-improvements.md
./docs/architecture/design-proposals/dashboard/error-handling/error-handling.md
./docs/architecture/design-proposals/dashboard/application-view-revamp.md
./docs/architecture/design-proposals/third-party-add-repository.md
./docs/architecture/design-proposals/replacing-mongodb.md
./docs/architecture/design-proposals/sso-without-oidc-api-server.md

./docs/developer/dashboard.md
./docs/developer/pinniped-proxy.md
./docs/developer/kubeapps-apis.md
./docs/developer/assetsvc.md
./docs/developer/using-makefiles.md
./docs/developer/asset-syncer.md
./docs/developer/kubeops.md
./docs/developer/apprepository-controller.md
./docs/developer/issue-triage-process.md
./docs/developer/update-api-docs.md
./docs/developer/ci.md
./docs/developer/README.md
./docs/developer/release-process.md
./docs/developer/custom-form-component-support.md
./docs/developer/end-to-end-tests.md
./docs/developer/testing-environment.md
./docs/developer/translate-kubeapps.md
./docs/developer/custom-app-view-support.md
./docs/developer/build.md
./docs/developer/basic-form-support.md
```

**Comment by absoludity** [comment]
See https://diataxis.fr/ for a good overview and argument for documentation structure.

**Comment by ppbaena** [comment]
New structure proposed in #4184 :

| 1st level     | 2nd level                                                          | New Path                                                  | Current path                                                                                                                                                         | Action    | Comments                                                                                                       |
| ------------- | ------------------------------------------------------------------ | --------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------- | -------------------------------------------------------------------------------------------------------------- |
| Overview      |                                                                    | ./README.md                                               | ./README.md                                                                                                                                                          | Update    | Update to follow the new doc structure                                                                         |
| Background    |                                                                    |                                                           |                                                                                                                                                                      |           |                                                                                                                |
|               | Architecture                                                       | ./docs/background/architecture/architecture.md            | ./docs/architecture/overview.md                                                                                                                                      | Update    | To be updated to the new architecture.                                                                         |
|               | Kubeapps API documentation                                         | ./docs/background/update-api-docs.md                      | ./docs/developer/update-api-docs.md                                                                                                                                  | Keep      |                                                                                                                |
| Tutorials     |                                                                    | ./docs/tutorials/readme.md                                |                                                                                                                                                                      |           |                                                                                                                |
|               | Get started with Kubeapps                                          | ./docs/tutorials/getting-started.md                       | ./docs/user/getting-started.md                                                                                                                                       | Keep      |                                                                                                                |
|               | Using an OIDC provider                                             | ./docs/tutorials/using-an-OIDC-provider.md                | ./docs/user/using-an-OIDC-provider.md                                                                                                                                | Keep      |                                                                                                                |
|               | Kubeapps on VMware Tanzu™ Kubernetes Grid™                         | ./docs/tutorials/kubeapps-on-tkg/introduction.md          | ./docs/step-by-step/kubeapps-on-tkg/introduction.md                                                                                                                  | Keep      | Remove ./docs/step-by-step/kubeapps-on-tkg/README.md                                                           |
|               | /img                                                               | ./docs/tutorials/kubeapps-on-tkg/img/                     | ./docs/step-by-step/kubeapps-on-tkg/img/                                                                                                                             | Keep      |                                                                                                                |
|               | Configure an Identity Management Provider in a TKG cluster         | ./docs/tutorials/kubeapps-on-tkg/step-1.md                | ./docs/step-by-step/kubeapps-on-tkg/step-1.md                                                                                                                        | Keep      |                                                                                                                |
|               | Configure and Install Kubeapps                                     | ./docs/tutorials/kubeapps-on-tkg/step-2.md                | ./docs/step-by-step/kubeapps-on-tkg/step-2.md                                                                                                                        | Keep      |                                                                                                                |
|               | Add Application Repositories to Kubeapps                           | ./docs/tutorials/kubeapps-on-tkg/step-3.md                | ./docs/step-by-step/kubeapps-on-tkg/step-3.md                                                                                                                        | Keep      |                                                                                                                |
|               | Deploy and Manage Applications with Kubeapps                       | ./docs/tutorials/kubeapps-on-tkg/step-4.md                | ./docs/step-by-step/kubeapps-on-tkg/step-4.md                                                                                                                        | Keep      |                                                                                                                |
|               | Conclusion                                                         | ./docs/tutorials/kubeapps-on-tkg/conclusions.md           | ./docs/step-by-step/kubeapps-on-tkg/conclusion.md                                                                                                                    | Keep      |                                                                                                                |
|               | Managing Carvel Packages with Kubeapps                             | .docs/tutorials/managing-carvel-packages.md               | ./docs/user/managing-carvel-packages.md                                                                                                                              | Keep      |                                                                                                                |
|               | Managing Operators with Kubeapps                                   | ./docs/tutorials/operators.md                             | ./docs/user/operators.md                                                                                                                                             | Keep      |                                                                                                                |
|               |                                                                    |                                                           |                                                                                                                                                                      |           |                                                                                                                |
| How-to Guides |                                                                    | ./docs/howto/readme.md                                    |                                                                                                                                                                      |           |                                                                                                                |
|               | Dashboard                                                          |                                                           | ./docs/user/dashboard.md                                                                                                                                             | Deprecate | To be replaced by:<br>Add Application Repositories to Kubeapps<br>Deploy and Manage Applications with Kubeapps |
|               | Configure and Install Kubeapps                                     | ./docs/howto/config-kubeapps.md                           | ./docs/howto/kubeapps-on-tkg/step-2.md                                                                                                                               | Update    | New file based on these documentation for TKG                                                                  |
|               | Add Application Repositories to Kubeapps                           | ./docs/howto/app-repository.md                            | ./docs/howto/kubeapps-on-tkg/step-3.md                                                                                                                               | Update    | New file based on these documentation for TKG                                                                  |
|               | Using a Private Repository with Kubeapps                           | ./docs/howto/private-app-repository.md                    | ./docs/user/private-app-repository.md                                                                                                                                | Keep      |                                                                                                                |
|               | Deploy and Manage Applications with Kubeapps                       | ./docs/howto/kubeapps-on-tkg/step-4.md                    | ./docs/howto/kubeapps-on-tkg/step-4.md                                                                                                                               | Update    | New file based on these documentation for TKG                                                                  |
|               | Access control in Kubeapps                                         | ./docs/howto/access-control.md                            | ./docs/user/access-control.md                                                                                                                                        | Keep      |                                                                                                                |
|               | Configure OIDC providers                                           | ./docs/howto/OIDC/README.md                               |                                                                                                                                                                      | Update    | To link all different OIDC providers from the tutorial                                                         |
|               | Configure an OIDC provider with Pinniped                           | ./docs/howto/OIDC/using-an-OIDC-provider-with-pinniped.md | ./docs/user/using-an-OIDC-provider-with-pinniped.md                                                                                                                  | Keep      |                                                                                                                |
|               | Configuring VMware Cloud Services as an OIDC provider              | ./docs/howto/OIDC/OAuth2OIDC-VMware-cloud-services.md     | ./docs/user/OIDC/OAuth2OIDC-VMware-cloud-services.md                                                                                                                 | Keep      |                                                                                                                |
|               | Configuring Azure Active Directory as an OIDC provider             | ./docs/howto/OIDC/OAuth2OIDC-azure-active-directory.md    | ./docs/user/OIDC/OAuth2OIDC-azure-active-directory.md                                                                                                                | Keep      |                                                                                                                |
|               | Configuring Google OIDC as an OIDC provider                        | ./docs/howto/OIDC/OAuth2OIDC-google-openid-connect.md     | ./docs/user/OIDC/OAuth2OIDC-google-openid-connect.md                                                                                                                 | Keep      |                                                                                                                |
|               | Configuring Keycloak as an OIDC provider                           | ./docs/howto/OIDC/OAuth2OIDC-keycloak.md                  | ./docs/user/OIDC/OAuth2OIDC-keycloak.md                                                                                                                              | Keep      |                                                                                                                |
|               | Configuring Dex as an OIDC provider                                | ./docs/howto/OIDC/OAuth2OIDC-dex.md                       | ./docs/user/OIDC/OAuth2OIDC-dex.md                                                                                                                                   | Keep      |                                                                                                                |
|               | Deploying an auth proxy to access Kubeapps                         | ./docs/howto/OIDC/OAuth2OIDC-oauth2-proxy.md              | ./docs/user/OIDC/OAuth2OIDC-oauth2-proxy.md                                                                                                                          | Keep      |                                                                                                                |
|               | Debugging auth failures when using OIDC                            | ./docs/howto/OIDC/OAuth2OIDC-debugging.md                 | ./docs/user/OIDC/OAuth2OIDC-debugging.md                                                                                                                             | Update    | Add an index to improve UX                                                                                     |
|               | Configuring Kubeapps for multiple clusters                         | ./docs/howto/deploying-to-multiple-clusters.md            | ./docs/user/deploying-to-multiple-clusters.md                                                                                                                        | Keep      |                                                                                                                |
|               | Offline installation                                               | ./docs/howto/offline-installation.md                      | ./docs/user/offline-installation.md                                                                                                                                  | Keep      |                                                                                                                |
|               | Syncing App Repositories using a webhook                           | ./docs/howto/syncing-apprepository-webhook.md             | ./docs/user/syncing-apprepository-webhook.md                                                                                                                         | Keep      |                                                                                                                |
|               | Basic Form Support                                                 | ./docs/howto/basic-form-support.md                        | ./docs/developer/basic-form-support.md                                                                                                                               | Keep      |                                                                                                                |
|               | Custom Form Component Support                                      | ./docs/howto/custom-form-component-support.md             | ./docs/developer/custom-form-component-support.md                                                                                                                    | Keep      |                                                                                                                |
|               | Custom App View Support                                            | ./docs/howto/custom-app-view-support.md                   | ./docs/developer/custom-app-view-support.md                                                                                                                          | Keep      |                                                                                                                |
|               |                                                                    |                                                           |                                                                                                                                                                      |           |                                                                                                                |
| Reference     |                                                                    | ./docs/reference/readme.md                                |                                                                                                                                                                      | Update    |                                                                                                                |
|               | Engineering Decission Records                                      | ./docs/reference/edr/edr.md                               | ./docs/architecture/design-proposals/README.md                                                                                                                       | Update    | Grouped and ordered by date<br>design-proposals folder moved to reference and renamed                          |
|               | /img                                                               | ./docs/reference/edr/img/                                 | ./docs/architecture/design-proposals/img/                                                                                                                            | Keep      |                                                                                                                |
|               | Authentication (AuthN) and Authorization (AuthZ)                   |                                                           | ./docs/architecture/design-proposals/authentication-and-authorization.md                                                                                             | Keep      |                                                                                                                |
|               | OAuth 2 - OpenID Connect (OIDC) Support                            |                                                           | [https://docs.google.com/document/d/1YZzLtIbS2copQJgspFiMd0eAhpyt8u19MUSDEvH2X4g](https://docs.google.com/document/d/1YZzLtIbS2copQJgspFiMd0eAhpyt8u19MUSDEvH2X4g)   | Keep      |                                                                                                                |
|               | Supporting SSO without requiring K8s API server oidc configuration |                                                           | ./docs/architecture/design-proposals/sso-without-oidc-api-server.md                                                                                                  | Keep      |                                                                                                                |
|               | Dashboard Error Handling                                           |                                                           | ./docs/architecture/design-proposals/dashboard/error-handling/error-handling.md                                                                                      | Keep      |                                                                                                                |
|               | Simpler deployments                                                |                                                           | ./docs/architecture/design-proposals/dashboard/deployment-improvements.md                                                                                            | Keep      |                                                                                                                |
|               | Improve UX in application view                                     |                                                           | ./docs/architecture/design-proposals/dashboard/application-view-revamp.md                                                                                            | Keep      |                                                                                                                |
|               | Upgrade to Clarity design                                          |                                                           | ./docs/architecture/design-proposals/clarity-ui.md                                                                                                                   | Keep      |                                                                                                                |
|               | Notify Available Chart Updates                                     |                                                           | [https://docs.google.com/document/d/1oG9nerd5CurWSIwH33kKCsOCtkSFMgcm8SuTJyuSnxs/](https://docs.google.com/document/d/1oG9nerd5CurWSIwH33kKCsOCtkSFMgcm8SuTJyuSnxs/) | Keep      |                                                                                                                |
|               | Multicluster support                                               |                                                           | ./docs/architecture/design-proposals/multi-cluster-support.md                                                                                                        | Keep      |                                                                                                                |
|               | Operators support                                                  |                                                           | ./docs/architecture/design-proposals/operators-support-poc.md                                                                                                        | Keep      |                                                                                                                |
|               | Third party integration for adding a chart repository              |                                                           | ./docs/architecture/design-proposals/third-party-add-repository.md                                                                                                   | Keep      |                                                                                                                |
|               | Replacing MongoDB support                                          |                                                           | ./docs/architecture/design-proposals/replacing-mongodb.md                                                                                                            | Keep      |                                                                                                                |
|               | Helm 3 support                                                     |                                                           | ./docs/architecture/design-proposals/helm3.md                                                                                                                        | Keep      |                                                                                                                |
|               | OCI support Helm charts in Kubeapps                                |                                                           | ./docs/architecture/design-proposals/OCI-registries.md                                                                                                               | Keep      |                                                                                                                |
|               | Kubeapps Developer Guides                                          | ./docs/reference/developer/readme.md                      | ./docs/developer/README.md                                                                                                                                           | Update    |                                                                                                                |
|               | How to set up the environment using the provided makefile targets  | ./docs/reference/developer/using-makefiles.md             | ./docs/developer/using-makefiles.md                                                                                                                                  | Keep      |                                                                                                                |
|               | Kubeapps Build Guide                                               | ./docs/reference/developer/build.md                       | ./docs/developer/build.md                                                                                                                                            | Keep      |                                                                                                                |
|               | Kubeapps Dashboard Developer Guide                                 | ./docs/reference/developer/dashboard.md                   | ./docs/developer/dashboard.md                                                                                                                                        | Keep      |                                                                                                                |
|               | Kubeapps Pinniped-Proxy Developer Guide                            | ./docs/reference/developer/pinniped-proxy.md              | ./docs/developer/pinniped-proxy.md                                                                                                                                   | Keep      |                                                                                                                |
|               | Kubeapps APIs service                                              | ./docs/reference/developer/kubeapps-apis.md               | ./docs/developer/kubeapps-apis.md                                                                                                                                    | Keep      |                                                                                                                |
|               | Kubeapps asset-syncer Developer Guide                              | ./docs/reference/developer/asset-syncer.md                | ./docs/developer/asset-syncer.md                                                                                                                                     | Keep      |                                                                                                                |
|               | Kubeapps assetsvc Developer Guide                                  | ./docs/reference/developer/assetsvc.md                    | ./docs/developer/assetsvc.md                                                                                                                                         | Keep      |                                                                                                                |
|               | Kubeapps kubeops Developer Guide                                   | ./docs/reference/developer/kubeops.md                     | ./docs/developer/kubeops.md                                                                                                                                          | Update    | To be deprecated?                                                                                              |
|               | Kubeapps apprepository-controller Developer Guide                  | ./docs/reference/developer/apprepository-controller.md    | ./docs/developer/apprepository-controller.md                                                                                                                         | Keep      |                                                                                                                |
|               | Kubeapps release process                                           | ./docs/reference/developer/release-process.md             | ./docs/developer/release-process.md                                                                                                                                  | Keep      |                                                                                                                |
|               | Kubeapps issue triage process                                      | ./docs/reference/developer/issue-triage-process.md        | ./docs/developer/issue-triage-process.md                                                                                                                             | Keep      |                                                                                                                |
|               | Kubeapps Testing Guides                                            | ./docs/reference/testing/readme.md                        |                                                                                                                                                                      | Update    |                                                                                                                |
|               | Kubeapps CI configuration                                          | ./docs/reference/testing/ci.md                            | ./docs/developer/ci.md                                                                                                                                               | Keep      |                                                                                                                |
|               | Setup Kubeapps testing environment                                 | ./docs/reference/testing/testing-environment.md           | ./docs/developer/testing-environment.md                                                                                                                              | Keep      |                                                                                                                |
|               | End-to-end tests in Kubeapps                                       | ./docs/reference/testing/end-to-end-tests.md              | ./docs/developer/end-to-end-tests.md                                                                                                                                 | Keep      |                                                                                                                |
|               | Kubeapps Translation                                               | ./docs/reference/translate-kubeapps.md                    | ./docs/developer/translate-kubeapps.md                                                                                                                               | Keep      |                                                                                                                |

**Comment by ppbaena** [comment]
Sorry for the long table (need to use a Convert Spreadsheet to Markdown tool) :).

**Comment by ppbaena** [comment]
Can be closed after merging #4516
